### DISCUSSION TITLE
Safari on iOS graphics problem

### DISCUSSION THREAD

**Discussion by jtogle** [mozilla/hubs#2946] [General Questions & Discussion] [answered]
Graphics (selecting your avatar, then the contents of the room itself) refuse to load on my iPhone 8 Plus using Safari and running iOS 13.6.1. The UI loads, I get the happy "you've entered the room" sound, and my avatar shows up for others, but the viewport into the room is black (as was the small window where you select your avatar). Works perfectly fine on an iPad, Quest, desktop browser. I can load the cube in get.webgl.org so it isn't that. The only solution I could find was clearing web history and such. Did that to no avail. Anyone have an idea to save me from this sad nap?

**Comment by j-conrad** [comment]
I tested yesterday (9/8/2020) on iOS 13.6.1 (as well as 13.7) with an iPhone 11. Everything loaded fine, including avatars.

To narrow things down, it would help to provide which scene you're attempting to load. Maybe the default one?

**Comment by emclaren** [comment]
Hi @jtogle - I wanted to check if maybe it had something to do with the specific room you entered into. If you go to hubs.mozilla.com and click "create room" is the behaviour the same as you described in your question?

**Comment by jtogle** [comment]
Yes, it is pervasive and n the iPhone. The same room will load fine on an
iPad. It’s strange!

On Wed, Sep 9, 2020 at 8:38 PM elgin mclaren <notifications@github.com>
wrote:

>
>
> Hi @jtogle <https://github.com/jtogle> - I wanted to check if maybe it
> had something to do with the specific room you entered into. If you go to
> hubs.mozilla.com and click "create room" is the behaviour the same as you
> described in your question?
>
>
>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/hubs/discussions/2946#discussioncomment-75464>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEOETIMU5WNLU7UOJ6E27TTSFANZ7ANCNFSM4QPALPOQ>
> .
>
>
> --
Todd Ogle, Ph.D.

Executive Director, Applied Research in Immersive Environments and
Simulations
University Libraries
560 Drillfield Drive
Virginia Tech
Blacksburg, VA 24061
Ph: 540-231-1188

http://lib.vt.edu/
http://historyviz.com/
http://www.hci.vt.edu/

**Reply by emclaren** [reply]
I tested another iPhone (running a slightly older OS), and it seems to work fine. I will post in the community discord about this to see if it happens for anyone else. Thanks for reporting back!

**Reply by jtogle** [reply]
Thanks @emclaren! I found a few folks reporting the same or similar issue on Discord before I posted but didn't come across a clear resolution. I've seen it work on everthing from an iPhone 6 with out of date iOS to an iPhone 11 and an iPad, so it must be me!

**Comment by jtogle** [comment]
FWIW here are screenshots:









On Wed, Sep 9, 2020 at 9:05 PM elgin mclaren <notifications@github.com>
wrote:

>
>
> I tested another iPhone (running a slightly older OS), and it seems to
> work fine. I will post in the community discord about this to see if it
> happens for anyone else. Thanks for reporting back!
>
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/hubs/discussions/2946#discussioncomment-75477>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEOETINM3OKXUHHNR3H4DQTSFAQ6JANCNFSM4QPALPOQ>
> .
>
>
> --
Todd Ogle, Ph.D.

Executive Director, Applied Research in Immersive Environments and
Simulations
University Libraries
560 Drillfield Drive
Virginia Tech
Blacksburg, VA 24061
Ph: 540-231-1188

http://lib.vt.edu/
http://historyviz.com/
http://www.hci.vt.edu/

**Comment by jtogle** [comment]
Sorry, I attempted to include those screenshots via email. Here they are. The thing to notice is that it is just the content window that is black. The UI is there. Sound works. I just updated my phone (iPhone 8 Plus) to iOS 13.7 and no change. I suspect it is something with Safari but I've cleared the cache and so forth to no avail.

![Flawless Brilliant Outing  Hubs by Mozilla (2)](https://user-images.githubusercontent.com/18631073/92743655-be10eb80-f34e-11ea-843f-faf71eb7bc33.png)
![Flawless Brilliant Outing  Hubs by Mozilla (1)](https://user-images.githubusercontent.com/18631073/92743657-be10eb80-f34e-11ea-9f29-7b91f452e274.png)
![Flawless Brilliant Outing  Hubs by Mozilla](https://user-images.githubusercontent.com/18631073/92743660-be10eb80-f34e-11ea-8041-b540620b156b.png)

**Comment by emclaren** [comment]
Created a bug report for this issue -  https://github.com/mozilla/hubs/issues/3003

**Comment by emclaren** [comment, marked as answer]
@jtogle - I just did a search in Discord and one of our community members reported a similar issue just the other day. They reported:

"09/08/2020
I solved it. I had turned on 2 experimental features of Safari (WebGL 2.0 and WebGPU). When I turned those off and cleared the cache it worked fine."

Is it possible that you have these features or something similar enabled?

**Comment by jtogle** [comment]
Yes, that did it! Thank you so much! I switched to watching Git for
solutions and hadn't checked back in on Discord. Thank you!!!

On Thu, Sep 10, 2020 at 4:33 PM elgin mclaren <notifications@github.com>
wrote:

> @jtogle <https://github.com/jtogle> - I just did a search in Discord and
> one of our community members reported a similar issue in our Discord just
> the other day. They reported:
>
> "09/08/2020
> I solved it. I had turned on 2 experimental features of Safari (WebGL 2.0
> and WebGPU). When I turned those off and cleared the cache it worked fine."
>
> Is it possible that you have these features or something similar enabled?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/hubs/discussions/2946#discussioncomment-76491>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEOETIIBUXGOM7NCYCO3I63SFEZY5ANCNFSM4QPALPOQ>
> .
>


-- 
Todd Ogle, Ph.D.

Executive Director, Applied Research in Immersive Environments and
Simulations
University Libraries
560 Drillfield Drive
Virginia Tech
Blacksburg, VA 24061
Ph: 540-231-1188

http://lib.vt.edu/
http://historyviz.com/
http://www.hci.vt.edu/

**Reply by emclaren** [reply]
nice!

**Reply by jtogle** [reply]
I really appreciate you tracking this down!
